tempesta_fw/t/functional/helpers/deproxy.py

```diff
@@ -212,20 +212,26 @@ def __init__(self, message_text=None, body_parsing=True, body_void=False):
 
     def parse_text(self, message_text, body_parsing=True):
         self.body_parsing = body_parsing
-        self.msg = message_text
-        stream = StringIO(self.msg)
+        stream = StringIO(message_text)
         self.__parse(stream)
+        self.__set_str_msg()
 
     def __parse(self, stream):
         self.parse_firstline(stream)
         self.parse_headers(stream)
         self.body = ''
         self.parse_body(stream)
 
+    def __set_str_msg(self):
+        self.msg = '\r\n'.join([self.get_firstline(), str(self)])
```

Comment:
The patch set fixes quite a lot of dirty code in the test framework. I would like to ask for one more cleanup.
For both request and response you have to join firstline and the rest of the message in many places: in `__set_str_msg()`, in `[Request|Responce].update()`. But in the same time `__str__()` method ignores firstline when it shouldn't.

To make the code looks better we should join the firstline to the rest of the message in `__str__()` method. After that we won't need to pass the first line to `update()` method, so both Request and Response won't need to override the `update()` method. `__set_str_msg()`  will be changed to simple `self.msg = str(self)`.

Reply:
Corrected.